packages-ts/starknet-gauntlet-ocr2/src/commands/ocr2/deploy.ts

```diff
@@ -20,13 +20,29 @@ type ContractInput = [
 
 const makeUserInput = async (flags, args, env): Promise<UserInput> => {
   if (flags.input) return flags.input as UserInput
+
+  if (flags.rdd) {
+    const rdd = getRDD(flags.rdd)
+    const contractAddr = args[0]
+    const aggregator = rdd[CONTRACT_TYPES.AGGREGATOR][contractAddr]
+    return {
+      maxAnswer: aggregator.maxSubmissionValue,
+      minAnswer: aggregator.minSubmissionValue,
+      decimals: aggregator.decimals,
+      description: aggregator.name,
+      billingAccessController: aggregator.billingAccessController,
+      linkToken: aggregator.linkToken,
+      owner: aggregator.owner || env.account,
+    }
+  }
+
```

Comment:
need to use `getRDD` in `makeUserInput` rather than `makeContractInput` as to capture the contract addr arg and pass input validation from gauntlet-core

Reply:
manually tested to be all working with a local devnet, but noting that the rdd files from [this PR as an example](https://github.com/smartcontractkit/reference-data-directory/pull/2719/files#diff-b74a33dc406f1252576900011901b9afce87b90c27a398cf46ea2421b9616d77) are missing a few required fields: `owner`, `linkToken`, `billingAccessController` (not sure if these are intended to be passed as env vars instead?)

Reply:
1. Yes, all RDD parsing should happen when handling user input in `makeUserInput`.
2. Those fields are to be passed with either flags or env vars.

Reply:
thanks @sdrug, seems a bit inconsistent in the way its being used, but ill add support so that these fields can be defined by either RDD, flags, or env vars (in that order of priority)

Reply:
Yeah I switched it over from `makeUserInput` to  `makeContractInput` where possible since it seemed more reasonable (we're constructing a separate contract input, not manipulating user input) but we can use `makeUserInput` where not possible